File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        nums_dic, cnt, pref = {},0,0
        nums_dic[0]=1
        for num in nums:
            pref+=num
            if pref not in nums_dic.keys():
                nums_dic[pref]=0
            if pref-k in nums_dic.keys():
                cnt+=nums_dic[pref-k]
            nums_dic[pref]+=1
            
        return cnt
            

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> hashMap;
        int prefix =0;
        int count =0;
        int n= nums.size();
        //1 1 1 
        //2
        //2
        hashMap[0]=1;
        for(int i=0;i<n;i++)
        {
            prefix += nums[i];
            if(hashMap.find(prefix - k) != hashMap.end())
                count += hashMap[prefix-k];
            hashMap[prefix]++;
        }
        return count;
    }
};